libs/core/src/lib/status-indicator/status-indicator.component.ts

```diff
@@ -0,0 +1,372 @@
+     * For default Object Status omit this property
+     */
+    @Input()
+    status: StatusIndicatorColor;
+
+    /** Define the text content of the Status indicator*/
+    @Input()
+    statusLabel: string;
+
+    /**
+     * positioning of the status indicator image withing the defined height and width .
+     */
+    @Input()
+    viewBox: string;
+
+    /**
+     * boolean value to define requirement of the label.
+     */
+    @Input()
+    hasLabel: boolean;
+
+    /**
+     * boolean value to be marked as a clickable
+     */
+    @Input()
+    clickable: boolean;
+
+    /** Aria label for the Status Indicator. */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Aria defines role description for the Status Indicator. */
+    @Input()
+    ariaRoledescription: string = null;
+
+    /** Aria Focusable for the Status Indicator. */
+    @Input()
+    focusable: boolean;
+
+    /** Aria Role for the Status Indicator. */
+    @Input()
+    role: string = null;
+
+    /** Aria Value Text for the Status Indicator. */
+    @Input()
+    ariaValuetext: string = null;
+
+    /** Aria title for the status indicator. */
+    @Input()
+    title: string = null;
+
+    /** defines the label position the value can be 'left' | 'right' | 'top' | 'bottom' */
+    @Input()
+    labelPosition: LablePosition;
+
+    @Input()
+    path: string[];
+
+    /**
+     * Offset value to be filled under the give percentatge value.
+     */
+    @Input()
+    fillPercentage: number;
+
+    /**
+     * value to define fill direction
+     */
+    @Input()
+    fillDirection: FillingDirection = 'clock';
+
+    get _fillDirection(): FillingDirection {
+        return this.fillDirection;
+    }
+
+    set _fillDirection(direction: FillingDirection) {
+        this.fillDirection = direction;
+    }
+
+    @ViewChild('maskTemplate')
+    maskTemplate: TemplateRef<any>;
+
+    /**
+     * FillingType to represent the fill pattern of the component
+     */
+    @Input()
+    fillingType: FillingType = 'lineardown';
+
+    /** represent the degree of angle to project the filling of the component */
+    @Input()
+    angle: number;
+
+    /** @hidden */
+    fillCalculator: number;
+    /** @hidden */
+    binaryString: string;
+    /** @hidden */
+    x1: string;
+    /** @hidden */
+    y1: string;
+    /** @hidden */
+    x2: string;
+    /** @hidden */
+    y2: string;
+    /** @hidden */
+    pointsArray: any[] = [];
+
+    /** @hidden */
+    constructor(private _elementRef: ElementRef<HTMLElement>, private _cd: ChangeDetectorRef) {}
+    ngAfterViewInit(): void {
+        this.angleCalculation();
+        this._cd.detectChanges();
+    }
+
+    /** @hidden */
+    ngOnChanges(): void {
+        this.calculateFilling();
+    }
+
+    /** @hidden */
+    elementRef(): ElementRef<HTMLElement> {
+        return this._elementRef;
+    }
+
+    /** @hidden */
+    private calculateFilling(): void {
```

Comment:
please add `_` to the all private methods
```suggestion
    private _calculateFilling(): void {
```